socorro/external/es/super_search_fields.py

```diff
@@ -1718,9 +1708,8 @@ def apply_schema_properties(fields, schema):
         "storage_mapping": {
             "fields": {
                 "full": {"index": "not_analyzed", "type": "string"},
-                "os_name": {"type": "string"},
```

Comment:
this is the only multi-field where in_database_name doesn't match the field name

Reply:
The comment on line 1716 suggests this shouldn't be analyzed. However, the values are things like "Windows NT", "Mac OS X", etc and people don't search with those values--they're searching with "Windows" and "Mac". Further, the Super Search form values don't match the actual values, either.

I think you've got the right idea here in indexing the value with the string analyzer. We should remove my comment on line 1706.

Then I wondered whether we should change the `data_validation_type` and `query_type` values. Fun fact, for all supersearch fields, those match with the exception of `addons_checked`. That has a `query_type` of `enum` but that should be `bool`.

Weirdly, the `query_type` affects whether we can use the field for histograms and `data_validation_type` affects the operators you can use (which I would have expected `query_type` to be used for) and how the querystring parameter values are converted.

I think given that `data_validation_type` and `query_type` match for all the fields, we should merge those into a single `data_type` field used for all the things. That'll simplify the code a little more and make it easier to read and understand the search code.

We could do that work (merging `data_validation_type` and `query_type` fields and fixing `addons_checked`) in a separate bug/PR. Let me know if you want me to write that up.

Reply:
yeah, i'd definitely prefer to do that in a separate PR